test/jdk/com/sun/jndi/dns/ConfigTests/TimeoutWithEmptyDatagrams.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.net.URIBuilder;
+
+import javax.naming.CommunicationException;
+import javax.naming.Context;
+import javax.naming.directory.InitialDirContext;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.SocketTimeoutException;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/*
+ * @test
+ * @bug 8339538
+ * @summary Tests that DnsClient correctly calculates left timeout in
+ *          presence of empty datagram packets.
+ * @library ../lib /test/lib
+ * @modules java.base/sun.security.util
+ * @run main/othervm TimeoutWithEmptyDatagrams
+ */
+
+public class TimeoutWithEmptyDatagrams extends DNSTestBase {
+    // initial timeout = 1/4 sec
+    private static final int TIMEOUT = 250;
+    // try 5 times per server
+    private static final int RETRIES = 5;
+    // DnsClient retries again with increased timeout if left
+    // timeout is less than this value, and max retry attempts
+    // is not reached
+    private static final int DNS_CLIENT_MIN_TIMEOUT = 50;
+
+    private long startTime;
```

Comment:
startTime does not need to be an instance variable: consider removing it and make it a local variable in the runTest() method.

Reply:
Thanks Daniel, made it locat variable in the `runTest` method: 11d0f3d038360c04710d78bd8cd036743f69fe41